Gemfile

```diff
@@ -34,6 +34,7 @@ gem 'jbuilder', '~> 2.5'
 gem 'omniauth'
 gem 'omniauth-fitbit'
 gem "figaro"
+gem 'fitbit-api-client', require: 'fitbit'
```

Comment:
I don't want to use this gem in this project. Looking at the source of this gem, it's questionable as to whether it is still supported. There does not seem to be a community around the gem (6 stars + 4 forks at the time of this writing). 

There are no tests written against its source code and I think I've found a defect in its implementation: The method call you're using - heart_rate_intraday_time_series - returns results for the non-intraday version of the api method.

Finally, I messed around with fetching data from the fitbit api endpoints and ran into authorization problems. The only way I was able to solve them was to delete the user account I created in this app and re-create it. I think there's an issue with refreshing the authorization tokens.

Please look into another gem that can perform the functions we need. In the absence of a better gem, let's investigate how we can improve this client (but I'd rather not have to do that).

For testing purposes, I was using this bit of code. I'd run it once and then attempt to run the code again an hour later (after the token expired):

```ruby
u = User.last
hrd = u.heart_rate_datas.first
d = (Date.today - 5).to_s
c = hrd.send(:client)
c.send(:get, "#{Fitbit::API_URI}/user/-/activities/heart/date/#{d}/1d/1min/time/00:00/23:59.json")
```